File: backend/app/utils/avatar.py
def default_avatar(seed: str | None = None) -> str:
    import urllib.parse
    s = seed or "user"
    return f"https://api.dicebear.com/6.x/adventurer/svg?seed={urllib.parse.quote_plus(s)}&backgroundType=gradientLinear&radius=50"

File: frontend/src/utils/avatar.js
export function getAvatarUrl(seed){
  // Use DiceBear Avatars (adventurer neutral) for high-quality illustrated avatars
  const s = encodeURIComponent(seed || Math.random().toString(36).slice(2,10))
  return `https://api.dicebear.com/6.x/adventurer/svg?seed=${s}&backgroundType=gradientLinear&radius=50`;
}
